webapp/context.py

```diff
@@ -195,3 +199,23 @@ def sort_by_key_and_ordered_list(list_to_sort, obj_key, ordered_list):
         if item[obj_key] in ordered_list
         else len(ordered_list),
     )
+
+
+def add_rate_limiting(request_limit="100/hour"):
+    # Rate limit requests to protect from spamming
+    # To adjust this rate visit
+    # https://limits.readthedocs.io/en/latest/quickstart.html#examples
+    def decorator(fn):
+        def wrapper(*args, **kwargs):
+            limit = parse(request_limit)
+            rate_limit = fixed_window.hit(limit)
+            if not rate_limit:
+                return flask.abort(429, f"The rate limit is: {request_limit}")
+            return fn(*args, **kwargs)
+
+        # Prevent naming conflicts
+        # https://stackoverflow.com/questions/17256602/assertionerror-view-function-mapping-is-overwriting-an-existing-endpoint-functi
```

Comment:
Ignore if this is unrelated, but [functools.wraps](https://docs.python.org/3/library/functools.html#functools.wraps) can be used so the name of the function is not lost